.github/workflows/performance.yml

```diff
@@ -0,0 +1,62 @@
+name: Performance Tests
+
+on:
+  push:
+    branches:
+      - trunk
+      - 'feature/**'
+
+jobs:
+  # Runs the performance test suite.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Sets up Node.js.
+  # - Installs npm dependencies.
+  # - Logs the running Docker containers.
+  # - Builds WordPress to run from the `build` directory.
+  # - Starts the WordPress Docker container.
+  # - Install WordPress within the Docker container.
+  # - Run the Performance tests.
+  # - Ensures version-controlled files are not modified or deleted.
+  performance:
+    runs-on: ubuntu-latest
+    if: ${{ github.repository == '10up/wordpress-develop' || github.event_name == 'push' }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+
+      - name: Set up Node.js
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
+        with:
+          node-version-file: '.nvmrc'
+          cache: npm
+
+      - name: Install npm dependencies
+        run: npm ci
+
+      - name: Docker debug information
+        run: |
+          docker -v
+          docker-compose -v
+          docker-compose run --rm mysql mysql --version
+          docker-compose run --rm php php --version
+          docker-compose run --rm php php -m
+          docker-compose run --rm php php -i
+          docker-compose run --rm php locale -a
```

Comment:
I'm unsure if these will require the docker environment to be started before running. Perhaps worth a test.

Reply:
@joemcgill Good point; in all existing core workflows where these commands are run the environment has already been started.

I would advise to change this to be aligned with how it is done in the `phpunit-tests.yml` and `test-coverage.yml` workflows:
1. Docker debug information
2. Start Docker environment (potentially after that also "Log running Docker containers"?)
3. WordPress Docker container debug information

Reply:
Thanks for the feedback. I have update steps as requested.